ipa/scripts/cli.py

```diff
@@ -325,6 +330,77 @@ def results(context, history_log):
         context.invoke(show, item=1)
 
 
+@click.command()
+@click.option(
+    '-c',
+    '--clear-log',
+    help='Clear the history log after archiving.',
+    is_flag=True
+)
+@click.option(
+    '-i',
+    '--items',
+    help='List of history items to archive. Must be a comma separated list.',
+    type=click.STRING
+)
+@click.argument(
+    'path',
+    type=click.Path(exists=True),
+)
+@click.pass_context
+def archive(context, clear_log, items, path):
+    """
+    Archive the history log and all results/log files.
+
+    After archive is created optionally clear the history log.
+    """
+    history_log = context.obj['history_log']
+
+    with open(history_log, 'r') as f:
+        # Get history items
+        history_items = f.readlines()
+
+    if items:
+        # Split comma separated list and cast indices to integer.
+        items = [int(item) for item in items.split(',')]
+
+        lines = []
+        for index in items:
+            lines.append(history_items[len(history_items) - index])
+        history_items = lines
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        for item in history_items:
+            # Copy log and results file,
+            # update results file with relative path.
+            archive_history_item(item, temp_dir)
+
+        time_stamp = datetime.now().strftime('%Y%m%d%H%M%S')
+        file_name = ''.join(['ipa_', time_stamp, '.tar.gz'])
+        archive_path = os.path.join(path, file_name)
+
```

Comment:
I think the user should have the option to give the archive a name that is meaningful to them rather than using the date based naming scheme in this case.